pastorsj/pandora-clone
# Queues and Likes [Breaking Changes]

## Acceptance Criteria
* User is able to like a song (does not affect their current playlist)
* Redis stores a queue of songs
* Slightly better error checking
* Everything else should not have regressed

## How To Test
* Make sure that your [Redis](https://redis.io) and [Neo4j](https://neo4j.com) databases are running. 
* Run the following commands to play a random playlist (feel free to confirm the pause, resume, and stop commands still work)
```
$ register
$ play
$ pause
$ resume
$ stop
```
* Run the genre command to get a list of genres
```
$ genres
```
* Run the following commands to play a playlist with a particular genre. After the `play genre`, you have to type what genre you want to play. In this case, rock was selected.
```
$ play genre
rock
```
The play genre is not on another thread, like play is, so we can't control the volume. I am not concerned about that, since the android app will deal with that.

## Notes
* To start the redis server, run
```
redis-server
```
* To start the neo4j database, run
```
neo4j start
```
* The `connect` command has been completely dropped from the client
* Every route besides the login, register, and genres route is protected
